lib/improver/utilities/spatial.py

```diff
@@ -41,6 +41,41 @@
 MAX_DISTANCE_IN_GRID_CELLS = 500
 
 
+def check_if_grid_is_equal_area(cube):
+    """Identify whether the grid is an equal area grid.
+    If not, raise an error.
+    Parameters
+    ----------
+    cube : Iris.cube.Cube
+        Cube with coordinates that will be cMAhecked.
+    Raises
+    ------
+    ValueError : Invalid grid: projection_x/y coords required
+    ValueError : Intervals between points along the x and y axis vary.
+                  Therefore the grid is not an equal area grid.
+    """
+    try:
+        for coord_name in ['projection_x_coordinate',
+                           'projection_y_coordinate']:
+            cube.coord(coord_name)
+    except CoordinateNotFoundError:
+        raise ValueError("Invalid grid: projection_x/y coords required")
+    for coord_name in ['projection_x_coordinate',
+                       'projection_y_coordinate']:
+        if np.sum(np.diff(np.diff(cube.coord(coord_name).points))) > 0:
+            msg = ("Intervals between points along the {} axis vary."
+                   "Therefore the grid is not an equal area grid."
+                   ).format(coord_name)
+            raise ValueError(msg)
+    x_mean = np.mean(np.diff(cube.coord("projection_x_coordinate").points))
```

Comment:
If you've already checked that they have the same resolution all the way along, you only need to check one point in the diff array not the mean.